src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -70,7 +63,7 @@ namespace xclhwemhal2 {
         std::unique_ptr<response_packet_info> response_header;
 	    size_t  i_len;
 	    size_t  ri_len;
-        unix_socket* Q2h_sock;
+        std::shared_ptr<unix_socket> Q2h_sock;
```

Comment:
Is this shared ownership, or is it exclusive?  If latter then use std::unique_ptr.

Reply:
Why not just using a value here and just pass references on it if there is no real concept of sharing?

Reply:
I have looked at the implementation and it seems like a simple type and makes more sense to have a move-only type and just use it by reference.